### PR TITLE
Started addings docs

### PR DESCRIPTION
**Pull request by k-boyle**

### CODE REVIEW

File: Core/src/main/java/kb/octane/core/CommandHandler.java [hunk 1]
```diff
@@ -29,24 +29,44 @@
...
+     * @param <T> The type of context that's used in commands.
+     * @return A new CommandHandler builder.
+     */
     public static <T extends CommandContext> Builder<T> builderForContext(Class<T> contextClazz) {
+        Preconditions.checkNotNull(contextClazz, "contextClazz cannot be null");
         return new Builder<>(contextClazz);
     }
 
+    /**
+     * Attempts to executes a command based on the given input.
```
Comment (trinitrotoluene) [inline Core/src/main/java/kb/octane/core/CommandHandler.java:65]: Attempts to **execute** a command [...]

File: Core/src/main/java/kb/octane/core/CommandHandler.java [hunk 2]
```diff
@@ -162,42 +205,71 @@ private Builder(Class<T> contextClazz) {
...
+         * @return The builder.
+         */
         public Builder<T> withBeanProvider(BeanProvider beanProvider) {
             Preconditions.checkNotNull(beanProvider, "beanProvider cannot be null");
             this.beanProvider = beanProvider;
             return this;
         }
 
+        /**
+         * Sets the ArgumentParser that will be used, if none if specified then the default will be used.
```
Comment (trinitrotoluene) [inline Core/src/main/java/kb/octane/core/CommandHandler.java:246]: if none **is** specified [...]

File: Core/src/main/java/kb/octane/core/module/Command.java [hunk 3]
```diff
@@ -77,10 +85,16 @@ public PreconditionResult runPreconditions(CommandContext context) {
             : SuccessfulPreconditionResult.get();
     }
 
+    /**
+     * @return The commands name.
```
Comment (trinitrotoluene) [inline Core/src/main/java/kb/octane/core/module/Command.java:89]: The command's name

File: Core/src/main/java/kb/octane/core/module/Command.java [hunk 4]
```diff
@@ -77,10 +85,16 @@ public PreconditionResult runPreconditions(CommandContext context) {
...
+     */
     public String name() {
         return name;
     }
 
+    /**
+     * @return The commands aliases.
```
Comment (trinitrotoluene) [inline Core/src/main/java/kb/octane/core/module/Command.java:96]: The command's aliases

File: Core/src/main/java/kb/octane/core/CommandHandler.java [hunk 5]
```diff
@@ -162,42 +205,71 @@ private Builder(Class<T> contextClazz) {
...
+         * @param argumentParser The argument parser.
+         * @return The bulder.
```
Comment (Raunow) [inline Core/src/main/java/kb/octane/core/CommandHandler.java:248]: The builder

File: Core/src/main/java/kb/octane/core/CommandHandler.java (hunk 2 above)
Comment (Raunow) [inline Core/src/main/java/kb/octane/core/CommandHandler.java:246]: if none is*

**Pull request merged**